tests/providers/amazon/aws/hooks/test_redshift.py

```diff
@@ -103,3 +107,67 @@ def test_cluster_status_returns_available_cluster(self):
         hook = RedshiftHook(aws_conn_id='aws_default')
         status = hook.cluster_status('test_cluster')
         assert status == 'available'
+
+
+class TestRedshiftSQLHookConn(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.connection = Connection(
+            conn_type='redshift', login='login', password='password', host='host', port=5439, schema="dev"
+        )
+
+        self.db_hook = RedshiftSQLHook()
+        self.db_hook.get_connection = mock.Mock()
+        self.db_hook.get_connection.return_value = self.connection
+
+    def test_get_uri(self):
+        expected = 'redshift+redshift_connector://login:password@host:5439/dev'
+        x = self.db_hook.get_uri()
+        assert x == expected
+
+    @mock.patch('airflow.providers.amazon.aws.hooks.redshift.redshift_connector.connect')
+    def test_get_conn(self, mock_connect):
+        self.db_hook.get_conn()
+        mock_connect.assert_called_once_with(
+            user='login', password='password', host='host', port=5439, database='dev'
+        )
+
+    @mock.patch('airflow.providers.amazon.aws.hooks.redshift.redshift_connector.connect')
+    def test_get_conn_extra(self, mock_connect):
+        self.connection.extra = json.dumps(
+            {
+                "iam": True,
+                "cluster_identifier": "my-test-cluster",
+                "profile": "default",
+            }
+        )
+        self.db_hook.get_conn()
+        mock_connect.assert_called_once_with(
+            user='login',
+            password='password',
+            host='host',
+            port=5439,
+            cluster_identifier="my-test-cluster",
+            profile="default",
+            database='dev',
+            iam=True,
+        )
+
+    @parameterized.expand(
+        [
+            ({}, {}, {}),
+            ({"login": "test"}, {}, {"user": "test"}),
+            ({}, {"user": "test"}, {"user": "test"}),
+            ({"login": "original"}, {"user": "overridden"}, {"user": "overridden"}),
+            ({"login": "test1"}, {"password": "test2"}, {"user": "test1", "password": "test2"}),
+        ],
+    )
+    @mock.patch('airflow.providers.amazon.aws.hooks.redshift.redshift_connector.connect')
+    def test_get_conn_overrides_correctly(self, test_args, test_kwargs, expected_call_args, mock_connect):
```

Comment:
```suggestion
    def test_get_conn_overrides_correctly(self, conn_params, conn_extra, expected_call_args, mock_connect):
```
hard to name this perfectly but test_args and test_kwargs... they're all "keywords args" but mainly the diff here is `extra` vs other conn params.  And this just seems a tad more clear.

Reply:
good point -- i've fixed this in 3821f40